sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/BlockBlobInputOutputStreamTest.groovy

```diff
@@ -147,30 +148,308 @@ class BlockBlobInputOutputStreamTest extends APISpec {
-        outStream.write(randomBytes, 1 * Constants.MB, 5 * Constants.MB)
-        outStream.close()
+        bc.upload(new ByteArrayInputStream(randomBytes), length, true)
 
-        when:
+        when: "Reading after etag has been changed"
         inputStream.read()
 
         then:
         thrown(IOException)
     }
+
+    def "IS consistent read control none"() {
+        setup:
+        int length = 6 * Constants.MB
+        byte[] randomBytes = getRandomByteArray(length)
+        bc.upload(new ByteArrayInputStream(randomBytes), length, true)
+
+        // Create the input stream and read from it.
+        // Note: Setting block size to 1 is inefficient but helps demonstrate the purpose of this test.
+        def inputStream = bc.openInputStream(new BlobInputStreamOptions().setBlockSize(1).setConsistentReadControl(ConsistentReadControl.NONE))
+        inputStream.read()
+
+        // Modify the blob again.
+        bc.upload(new ByteArrayInputStream(randomBytes), length, true)
+
+        when:
+        inputStream.read()
+
+        then: "Exception should not be thrown even though blob was modified"
+        notThrown(IOException)
+    }
+
+    def "IS consistent read control etag client chooses etag"() {
+        setup:
+        int length = Constants.KB
+        byte[] randomBytes = getRandomByteArray(length)
+        def blobContainerClient = versionedBlobServiceClient.createBlobContainer(containerName)
+        def blobClient = blobContainerClient.getBlobClient(generateBlobName())
+        blobClient.upload(new ByteArrayInputStream(randomBytes), length, true)
+
+        when:
+        // No eTag specified - client will lock on latest one.
+        def inputStream = blobClient.openInputStream(new BlobInputStreamOptions().setConsistentReadControl(ConsistentReadControl.ETAG))
+        def outputStream = new ByteArrayOutputStream()
+
+        then: "Successful read"
+        def b
+        while ((b = inputStream.read()) != -1) {
+            outputStream.write(b)
+        }
+
+        byte[] randomBytes1 = outputStream.toByteArray()
+        assert randomBytes1 == randomBytes
+    }
+
+    def "IS consistent read control etag user provides etag"() {
+        setup:
+        int length = Constants.KB
+        byte[] randomBytes = getRandomByteArray(length)
+        def blobContainerClient = versionedBlobServiceClient.createBlobContainer(containerName)
+        def blobClient = blobContainerClient.getBlobClient(generateBlobName())
+        blobClient.upload(new ByteArrayInputStream(randomBytes), length, true)
+        def properties = blobClient.getProperties()
+
+        when:
+        def inputStream = blobClient.openInputStream(new BlobInputStreamOptions().setConsistentReadControl(ConsistentReadControl.ETAG)
+        // User provides eTag to use
+            .setRequestConditions(new BlobRequestConditions().setIfMatch(properties.getETag())))
+        def outputStream = new ByteArrayOutputStream()
+
+        then: "Successful read"
```

Comment:
I think we should try modifying and check for failure on these after we get success to make sure that it's actually getting set.

Reply:
I see there's an error case below now.